### DISCUSSION TITLE
Breaking changes and versioning

### DISCUSSION THREAD

**Discussion by thesoulless** [libvips/libvips#3989] [General]
I'm a bit confused how libvips versioning works.

[Add "keep" flag to savers](https://github.com/libvips/libvips/pull/3585) was merged and released in v8.15, which adds a great unified flag to all the savers. But if I'm not mistaken strip was never marked as deprecated, and so removing it seems like a breaking change.

Or maybe the savers stability is not considered in the versioning?

**Comment by jcupitt** [comment]
Hello @thesoulless,

I think `strip` has been tagged as deprecated, hasn't it? If I run:

```
$ vips jpegsave
save image to jpeg file
usage:
   jpegsave in filename [--option-name option-value ...]
where:
   in           - Image to save, input VipsImage
   filename     - Filename to save to, input gchararray
optional arguments:
   Q            - Q factor, input gint
			default: 75
			min: 1, max: 100
   optimize-coding - Compute optimal Huffman coding tables, input gboolean
			default: false
   interlace    - Generate an interlaced (progressive) jpeg, input gboolean
			default: false
   trellis-quant - Apply trellis quantisation to each 8x8 block, input gboolean
			default: false
   overshoot-deringing - Apply overshooting to samples with extreme values, input gboolean
			default: false
   optimize-scans - Split spectrum of DCT coefficients into separate scans, input gboolean
			default: false
   quant-table  - Use predefined quantization table with given index, input gint
			default: 0
			min: 0, max: 8
   subsample-mode - Select chroma subsample operation mode, input VipsForeignSubsample
			default enum: auto
			allowed enums: auto, on, off
   restart-interval - Add restart markers every specified number of mcu, input gint
			default: 0
			min: 0, max: 2147483647
   keep         - Which metadata to retain, input VipsForeignKeep
			default flags: exif:xmp:iptc:icc:other:all
			allowed flags: none, exif, xmp, iptc, icc, other, all
   background   - Background value, input VipsArrayDouble
   page-height  - Set page height for multipage save, input gint
			default: 0
			min: 0, max: 10000000
   profile      - Filename of ICC profile to embed, input gchararray
operation flags: sequential nocache 
```

The new `keep` options is there, but `strip` has vanished. You can still use it (for compatibility), but it's no longer documented.

```
$ vips copy k2.jpg x.jpg[strip]
$
```

**Reply by thesoulless** [reply]
So it was marked as deprecated in the same release, and kept around for compatibility, I see.

My script was checking the `vips jpegsave` output and throw a waning since it couldn't find the `strip` flag anymore, but that's not how you mark a flag as deprecated in libvips.

Thanks for your time @jcupitt

**Reply by jcupitt** [reply]
That's right, it's the same in all the language bindings too -- the DEPRECATED flag on the argument means that it has vanished from all the pyvips docs and IDE completions, for example, but you can still use it.
